src/library/dlgtrackinfomulti.cpp

```diff
@@ -0,0 +1,833 @@
+
+namespace {
+
+const QString kVariousText = char('<') + QObject::tr("various") + char('>');
+const char* kOrigValProp = "origVal";
+const QString kClearItem = QStringLiteral("clearItem");
+
+/// If value differs from the current value, add it to the list.
+/// If new and current are identical, keep only one. Later on we can use the
+/// item count to maybe join the list and format the font accordingly.
+/// List can be a QStringList or a QList<double>
+template<typename T>
+void appendToTagList(QList<T>* pList, const T& value) {
+    if (pList->isEmpty() || !pList->contains(value)) {
+        pList->append(value);
+    }
+}
+
+void setItalic(QWidget* pEditor, bool italic) {
+    auto font = pEditor->font();
+    if (font.italic() == italic) {
+        return;
+    }
+    font.setItalic(italic);
+    pEditor->setFont(font);
+}
+
+void setBold(QWidget* pEditor, bool bold) {
+    auto font = pEditor->font();
+    if (font.bold() == bold) {
+        return;
+    }
+    font.setBold(bold);
+    pEditor->setFont(font);
+}
+
+/// Check if the text has been edited, i.e. is not <various>
+std::optional<QString> validEditText(QComboBox* pBox) {
+    QString origVal = pBox->property(kOrigValProp).toString();
+    if (pBox->currentIndex() == -1 &&
+            (pBox->lineEdit()->text() == origVal ||
+                    pBox->lineEdit()->placeholderText() == kVariousText)) {
+        // This is either a single-value box and the value changed, or this is a
+        // multi-value box and the placeholder text was removed when clearing it.
+        return std::nullopt;
+    }
+    // We have a new text
+    return pBox->currentText().trimmed();
+}
+
+/// Returns either the only value or the 'various' string.
+/// In the latter case make the text italic to indicate multiple values,
+/// also if only one is not empty.
+/// This is used for the Year and Track# fields as well as the sample rate and
+/// file ytpe labels. Optionally toggle bold (non-editable fields).
+/// pWidget can be a QLineEdit or a QPlainTextEdit (comment)
+QString getCommonValueOrVariousStringAndFormatFont(QWidget* pWidget,
+        QStringList& values,
+        bool toggleBold = false,
+        const QString& unit = QString()) {
+    if (values.size() == 1) {
+        setItalic(pWidget, false);
+        if (toggleBold) {
+            setBold(pWidget, true);
+        }
+        if (unit.isEmpty()) {
+            return values[0];
+        } else {
+            return values[0] + QChar(' ') + unit;
+        }
+    } else {
+        setItalic(pWidget, true);
+        if (toggleBold) {
+            setBold(pWidget, false);
+        }
+        return kVariousText;
+    }
+}
+
+} // namespace
+
+DlgTrackInfoMulti::DlgTrackInfoMulti(UserSettingsPointer pUserSettings)
+        // No parent because otherwise it inherits the style parent's
+        // style which can make it unreadable. Bug #673411
+        : QDialog(nullptr),
+          m_pUserSettings(std::move(pUserSettings)),
```

Comment:
std::move does not make sense here because this constructor is called with m_pConfig elsewhere, and that m_pConfig is still owned by the caller, so it cannot be moved into dlgtrackinfomulti.  in general, std::move is used in situations where the object being passed into the function call will *never again* be used by the caller.  This is not the case for the call pattern here, so I believe this std::move will have no effect, and it can be removed.  (UserSettingsPointer is just a shared pointer anyway, it is meant to be copied around and shared, not moved)

Reply:
I worked off DlgTrackInfo without questioning the basic structure, so I guess it can removed there as well.

Reply:
This does not move the ownership, because it is a shared pointer. It moves the shared pointer to avoid reference counting. So it is a valid approach. 

This follows the pattern to copy objects (in this case the shared pointer) by value parameters and then move them to the final destination.

Reply:
so you're moving the SharedPointer object from the constructor function parameter to the object being created... doesn't the parameter to the constructor go out of scope anyway, so there's no reference counting issue?  it still seems like overkill for such a light object, but ok, we can leave it.

Reply:
I agree that this is micro optimizing this case. 
But we have started to implement this pattern everywhere following the rule of thump.
* const ref for local read access
* value + move when kept as member 
* raw pointer, for write access

It would be interesting if the compiler is able to optimize the reference counting away without the move. I am afraid it cannot. 
The reference counting is not cheap, because memory fences are involved.